src/nni_manager/core/nniDataStore.ts

```diff
@@ -358,35 +339,94 @@ class NNIDataStore implements DataStore {
                     }
                 default:
             }
-            /* eslint-enable no-fallthrough */
             jobInfo.status = this.getJobStatusByLatestEvent(jobInfo.status, record.event);
+
             if (record.data !== undefined && record.data.trim().length > 0) {
-                const newHParam: any = this.parseHyperParameter(record.data);
-                if (newHParam !== undefined) {
-                    if (jobInfo.hyperParameters !== undefined) {
-                        let hParamIds: Set<number> | undefined = hParamIdMap.get(jobInfo.id);
-                        if (hParamIds === undefined) {
-                            hParamIds = new Set();
-                        }
-                        if (!hParamIds.has(newHParam.parameter_index)) {
-                            jobInfo.hyperParameters.push(JSON.stringify(newHParam));
-                            hParamIds.add(newHParam.parameter_index);
-                            hParamIdMap.set(jobInfo.id, hParamIds);
+                const metricData: any = this.parseMetricData(record.data);
```

Comment:
the `record.data` here is actually a hyperParameter stored using `storeTrialJobEvent`

Reply:
the data is as below, it called metric data, not only hyper parameters.

{"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 512, "lr": 0.01, "momentum": 0.738026583665898}, "parameter_index": 0}
--

Reply:
the above is a hyper parameter,  the content of parameters is generated by Tuner and other fields are meta-data,   set by nni sdk.

Reply:
Yes, but it was called as metricData... I didn't change it. And the actual metric data called MetricDataRecord.